clients/filesystem-fuse/src/default_raw_filesystem.rs

```diff
@@ -0,0 +1,398 @@
+            .await;
+        Ok(file_stat)
+    }
+
+    async fn read_dir(&self, file_id: u64) -> crate::filesystem::Result<Vec<FileStat>> {
+        let file_entry = self.get_file_entry(file_id).await?;
+        let mut child_filestats = self.fs.read_dir(&file_entry.path).await?;
+        for file in child_filestats.iter_mut() {
+            self.resolve_file_id_to_filestat(file, file.file_id).await;
+        }
+        Ok(child_filestats)
+    }
+
+    async fn open_file(&self, file_id: u64, flags: u32) -> crate::filesystem::Result<FileHandle> {
+        self.open_file_internal(file_id, flags, FileType::RegularFile)
+            .await
+    }
+
+    async fn open_dir(&self, file_id: u64, flags: u32) -> crate::filesystem::Result<FileHandle> {
+        self.open_file_internal(file_id, flags, FileType::Directory)
+            .await
+    }
+
+    async fn create_file(
+        &self,
+        parent_file_id: u64,
+        name: &str,
+        flags: u32,
+    ) -> crate::filesystem::Result<FileHandle> {
+        let parent_file_entry = self.get_file_entry(parent_file_id).await?;
+        let mut opened_file = self
+            .fs
+            .create_file(&parent_file_entry.path, name, OpenFileFlags(flags))
+            .await?;
+
+        opened_file.set_file_id(parent_file_id, self.next_file_id());
+
+        // insert the new file to file entry manager
+        {
+            let mut file_manager = self.file_entry_manager.write().await;
+            file_manager.insert(
+                parent_file_id,
+                opened_file.file_stat.file_id,
+                &opened_file.file_stat.path,
+            );
+        }
+
+        // put the file to the opened file manager
+        let opened_file = self.opened_file_manager.put(opened_file);
+        let opened_file = opened_file.lock().await;
+        Ok(opened_file.file_handle())
+    }
+
+    async fn create_dir(&self, parent_file_id: u64, name: &str) -> crate::filesystem::Result<u64> {
+        let parent_file_entry = self.get_file_entry(parent_file_id).await?;
+        let mut filestat = self.fs.create_dir(&parent_file_entry.path, name).await?;
+
+        filestat.set_file_id(parent_file_id, self.next_file_id());
+
+        // insert the new file to file entry manager
+        {
+            let mut file_manager = self.file_entry_manager.write().await;
+            file_manager.insert(filestat.parent_file_id, filestat.file_id, &filestat.path);
+        }
+        Ok(filestat.file_id)
+    }
+
+    async fn set_attr(&self, file_id: u64, file_stat: &FileStat) -> crate::filesystem::Result<()> {
+        let file_entry = self.get_file_entry(file_id).await?;
+        self.fs.set_attr(&file_entry.path, file_stat, true).await
+    }
+
+    async fn remove_file(&self, parent_file_id: u64, name: &str) -> crate::filesystem::Result<()> {
+        let parent_file_entry = self.get_file_entry(parent_file_id).await?;
+        self.fs.remove_file(&parent_file_entry.path, name).await?;
+
```

Comment:
should we check opened file when removing?

Reply:
If no error is reported, it must be successful. If you use `self.fs` for checks, it will affect performance.